game.php

```diff
@@ -714,233 +870,295 @@ function chess_is_draw_threefold_repetition(&$gamedata)
-		}
-	}
-
-	// Construct lists of white's and black's captured pieces.
-	$captured_pieces = array('white' => array(), 'black' => array());
-	foreach ($captured_pieces_all as $piece => $count) {
-		if ($count > 0) {
-			if (ctype_upper($piece)) {
-				$captured_pieces['white'] = array_merge($captured_pieces['white'], array_pad(array(), $count, $piece));
-			} elseif (ctype_lower($piece)) {
-				$captured_pieces['black'] = array_merge($captured_pieces['black'], array_pad(array(), $count, $piece));
-			}
-		}
-	}
-	#var_dump('captured_pieces_all', $captured_pieces_all);#*#DEBUG#
-	#var_dump('captured_pieces', $captured_pieces);#*#DEBUG#
-	$xoopsTpl->assign('chess_captured_pieces', $captured_pieces);
-
-	$xoopsTpl->assign('chess_pawn_promote_choices', $gamedata['fen_active_color'] == 'w' ? array('Q','R','B','N') : array('q','r','b','n'));
-
-	$xoopsTpl->assign('chess_allow_delete', chess_can_delete());
-
-	// popup window contents for selecting piece to which pawn is promoted
-	// (Note that template is compiled here by fetch(), so any template variables it uses must already be defined.)
-	$xoopsTpl->assign('chess_pawn_promote_popup', $user_color ? $xoopsTpl->fetch('db:chess_game_promote_popup.html') : '');
-
-	$xoopsTpl->assign('chess_ratings_enabled', chess_moduleConfig('rating_system') != 'none');
-
-	// security token
-	$xoopsTpl->assign('chess_xoops_request_token', is_object($GLOBALS['xoopsSecurity']) ? $GLOBALS['xoopsSecurity']->getTokenHTML() : '');
+    global $xoopsTpl;
+
+    $xoopsTpl->assign(
+        'xoops_module_header',
+        '
+		<link rel="stylesheet" type="text/css" media="screen" href="' . XOOPS_URL . '/modules/chess/assets/css/style.css">
+	'
+    );
+
+    $memberHandler = xoops_getHandler('member');
+
+    $white_user = $memberHandler->getUser($gamedata['white_uid']);
+
+    $white_username = is_object($white_user) ? $white_user->getVar('uname') : '?';
+
+    $black_user = $memberHandler->getUser($gamedata['black_uid']);
+
+    $black_username = is_object($black_user) ? $black_user->getVar('uname') : '?';
+
+    // Determine whether board is flipped (black at bottom) or "normal" (white at bottom).
+
+    switch ($orientation) {
+        default:
+        case _CHESS_ORIENTATION_ACTIVE:
+            $flip = 'b' == $gamedata['fen_active_color'];
+            break;
+        case _CHESS_ORIENTATION_WHITE:
+            $flip = false;
+            break;
+        case _CHESS_ORIENTATION_BLACK:
+            $flip = true;
+            break;
+    }
+
+    // Convert fen_piece_placement string into 8x8-array $tiles.
+
+    $tiles = [];
+
+    $ranks = explode('/', $gamedata['fen_piece_placement']);
+
+    if ($flip) {
+        $ranks = array_reverse($ranks);
+    }
+
+    foreach ($ranks as $rank) {
+        $expanded_row = preg_replace_callback(
+            '/(\d)/',
+
+            create_function(
+                '$matches',
+                'return str_repeat(\'x\', $matches[1]);'
+            ),
```

Comment:
Missing change from create_function to function from 1.08 Alpha 1
```suggestion
	    function ($matches) {
		return str_repeat('x', $matches[1]);
	    },
```